staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -65,8 +66,6 @@ func (w *realTimeoutFactory) TimeoutCh() (<-chan time.Time, func() bool) {
 // serveWatch will serve a watch response.
 // TODO: the functionality in this method and in WatchServer.Serve is not cleanly decoupled.
 func serveWatch(watcher watch.Interface, scope *RequestScope, mediaTypeOptions negotiation.MediaTypeOptions, req *http.Request, w http.ResponseWriter, timeout time.Duration, metricsScope string) {
-	defer watcher.Stop()
```

Comment:
What if we return somewhere in the middle of this func (on error)? Then the watch will not be stopped and will effectively leak...

Reply:
Done despite the change is a bit verbose